### PR TITLE
Adding HTTP status code for transformResponse

### PR DESCRIPTION
**Pull request by kaishuu0123**
# Related Issue

* refs #1214

# Motivation

* It would be more useful to get the HTTP status code as well
* I want to do conditional branching in the transformResponse function using the response status code.

# Note

* Let me know if I'm doing something wrong.

### CODE REVIEW

**Comment by kaishuu0123** [comment]
Sorry. Removed include changes to `dist/` in my pull request.

**Review by diegood — APPROVED** [review]

**Comment by jasonsaayman** [comment]
@kaishuu0123 can you please add some test cases for this

**Comment by kaishuu0123** [comment]
@jasonsaayman I have added one test case.
The test seems to fail, can you run the test again? (FormData TIMEOUT?)

```
Chrome Headless 101.0.4951.54 (Linux x86_64) FormData should allow FormData posting FAILED
	Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
```

File: test/specs/core/transformData.spec.js [hunk 1]
```diff
@@ -26,5 +26,28 @@ describe('core::transformData', function () {
...
+      'content-type': 'foo/bar',
+    };
+    var data = '';
+    data = transformData(data, headers, null, [function (data, headers) {
+      data += headers['content-type'];
+      return data;
+    }]);
+
+    expect(data).toEqual('foo/bar');
+  });
```
Comment (kaishuu0123) [inline test/specs/core/transformData.spec.js:41]: There was no test case that referenced headers, so a test case was added.

**Pull request merged**